src/python/zquantum/core/wip/estimators/estimation.py

```diff
@@ -182,26 +197,30 @@ def naively_estimate_expectation_values(
         backend: backend used for executing circuits
         estimation_tasks: list of estimation tasks
     """
-    circuits, operators, shots_per_circuit = zip(
-        *[(e.circuit, e.operator, e.number_of_shots) for e in estimation_tasks]
-    )
+    circuits, operators, shots_per_circuit = [], [], []
+    for task in estimation_tasks:
```

Comment:
I think it's better to avoid `append` if you can use list comprehension.
See [here](https://stackoverflow.com/questions/30245397/why-is-a-list-comprehension-so-much-faster-than-appending-to-a-list).

Reply:
Resolved here: d4222b2